frontend/src/components/core/Block/Block.tsx

```diff
@@ -15,733 +15,170 @@
-        return (
-          <ExceptionElement
-            width={width}
-            element={node.element.exception as ExceptionProto}
-          />
-        )
-
-      case "graphvizChart":
-        return (
-          <GraphVizChart
-            element={node.element.graphvizChart as GraphVizChartProto}
-            index={index}
-            width={width}
-          />
-        )
-
-      case "iframe":
-        return (
-          <IFrame element={node.element.iframe as IFrameProto} width={width} />
-        )
-
-      case "imgs":
-        return (
-          <ImageList
-            width={width}
-            element={node.element.imgs as ImageListProto}
-          />
-        )
-
-      case "json":
-        return <Json width={width} element={node.element.json as JsonProto} />
-
-      case "markdown":
-        return (
-          <Markdown
-            width={width}
-            element={node.element.markdown as MarkdownProto}
-          />
-        )
-
-      case "plotlyChart":
-        return (
-          <PlotlyChart
-            width={width}
-            height={height}
-            element={node.element.plotlyChart as PlotlyChartProto}
-          />
-        )
-
-      case "progress":
-        return (
-          <Progress
-            width={width}
-            element={node.element.progress as ProgressProto}
-          />
-        )
-
-      case "table":
-        return (
-          <Table element={node.immutableElement.get("table")} width={width} />
-        )
-
-      case "text":
-        return <Text width={width} element={node.element.text as TextProto} />
-
-      case "metric":
-        return <Metric element={node.element.metric as MetricProto} />
-
-      case "vegaLiteChart":
-        return (
-          <VegaLiteChart
-            element={node.immutableElement.get("vegaLiteChart")}
-            width={width}
-          />
-        )
-
-      case "video":
-        return (
-          <Video width={width} element={node.element.video as VideoProto} />
-        )
-
-      // Widgets
+  return child
+}
 
-      case "button": {
-        const buttonProto = node.element.button as ButtonProto
-        if (buttonProto.isFormSubmitter) {
-          const { formId } = buttonProto
-          const hasInProgressUpload = this.props.formsData.formsWithUploads.has(
-            formId
-          )
-          return (
-            <FormSubmitContent
-              element={buttonProto}
-              width={width}
-              hasInProgressUpload={hasInProgressUpload}
-              {...widgetProps}
-            />
-          )
+const ChildRenderer = (props: BlockProps): ReactElement => {
+  /** Recursively transform this BlockElement and all children to React Nodes. */
+  return (
+    <>
+      {props.node.children.map(
+        (node: ReportNode, index: number): ReactElement => {
+          if (node instanceof ElementNode) {
+            const childProps = assign({}, props, { node: node as ElementNode })
```

Comment:
Super nit but this feels like it would be easier to read as `const childProps = { ...props, node: node as ElementNode }`

Reply:
Done